src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -485,26 +471,6 @@ internal static PropertyValidationInfo GetPropertyValidationInfo(ModelBindingCon
                 value = dtoResult.Model;
             }
 
-            // 'Required' validators need to run first so that we can provide useful error messages if
-            // the property setters throw, e.g. if we're setting entity keys to null.
```

Comment:
Code was previously here to execute required validators before attempting to set the property, not before other validators. `DefaultObjectValidator` execution comes much later, after all the setting has settled.

With this change, the first error message will be from the `Exception` and not the required validator.